doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,845 @@
+        """
+        length = random.randint(1, 10)
+        letters = ''.join(random.choice(string.ascii_lowercase)
+                          for i in range(length))
+        return length, letters
+
+If the method yields its value:
+
+.. code-block:: python
+
+    def sample_values():
+        """
+        Generate an infinite sequence of random numbers.
+
+        The values are sampled from a continuous uniform distribution between
+        0 and 1.
+
+        Yields
+        ------
+        float
+            Random number generated.
+        """
+        while True:
+            yield random.random()
+
+.. _docstring.see_also:
+
+Section 5: See Also
+~~~~~~~~~~~~~~~~~~~
+
+This section is used to let users know about pandas functionality
+related to the one being documented. In rare cases, if no related methods
+or functions can be found at all, this section can be skipped.
+
+An obvious example would be the `head()` and `tail()` methods. As `tail()` does
+the equivalent as `head()` but at the end of the `Series` or `DataFrame`
+instead of at the beginning, it is good to let the users know about it.
+
+To give an intuition on what can be considered related, here there are some
+examples:
+
+* `loc` and `iloc`, as they do the same, but in one case providing indices and
+  in the other positions
+* `max` and `min`, as they do the opposite
+* `iterrows`, `itertuples` and `iteritems`, as it is easy that a user looking
+  for the method to iterate over columns ends up in the method to iterate
+  over rows, and vice-versa
+* `fillna` and `dropna`, as both methods are used to handle missing values
+* `read_csv` and `to_csv`, as they are complementary
+* `merge` and `join`, as one is a generalization of the other
+* `astype` and `pandas.to_datetime`, as users may be reading the documentation
+  of `astype` to know how to cast as a date, and the way to do it is with
+  `pandas.to_datetime`
+* `where` is related to `numpy.where`, as its functionality is based on it
+
+When deciding what is related, you should mainly use your common sense and
+think about what can be useful for the users reading the documentation,
+especially the less experienced ones.
+
+When relating to other libraries (mainly `numpy`), use the name of the module
+first (not an alias like `np`). If the function is in a module which is not
+the main one, like `scipy.sparse`, list the full module (e.g.
+`scipy.sparse.coo_matrix`).
+
+This section, as the previous, also has a header, "See Also" (note the capital
+S and A). Also followed by the line with hyphens, and preceded by a blank line.
+
+After the header, we will add a line for each related method or function,
+followed by a space, a colon, another space, and a short description that
+illustrated what this method or function does, why is it relevant in this
+context, and what are the key differences between the documented function and
+the one referencing. The description must also finish with a dot.
```

Comment:
I don't think we need to *require* a description, do we? I think if you're writing `read_csv` and want to link to `DataFrame.to_csv`, just the link should be sufficient.